src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const logs = [];
+  const stateClone = { ...state };
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(stateClone, extraData);
+        break;
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          delete stateClone[key];
+        }
+        break;
+      case 'clear':
+        for (const key in stateClone) {
```

Comment:
I have made the same mistake: took the code from the previous task.
And was told that this part should be written with an empty object.
Smth like this: stateClone = {};

But still, the job is done great!)

Reply:
Sounds reasonable, but probably not critical here